Dashboard/src/components/Github.js

```diff
@@ -3,17 +3,17 @@ import Panel from './common/Panel';
 import Container from './common/Container';
 import github from './assets/github.svg';
 import moment from 'moment';
-const COMPONENT_NAME = "github";
+const COMPONENT_NAME = "github"
 class Github extends Container {
   constructor(props) {
     super(props, COMPONENT_NAME);
   }
 
   render() {
     return (
-      <div>
+      <div className={this.constructor.name.toLocaleLowerCase()+ '-panel panel'}>
         <Panel
-          title={COMPONENT_NAME}
+          title={this.constructor.name}
```

Comment:
I was trying to get rid of the component_name variable all together and just use the name that is used by the class. I think I couldn't get rid of it because it was being used somewhere else and I wasn't sure how to replace it. Have to look into it further.

Reply:
I would argue it's less readable than using a string literal: `title = {'github'}` vs. `title= {this.constructor.name}`. The reason `COMPONENT_NAME` is being used instead of a literal is to avoid the [Magic string](https://en.wikipedia.org/wiki/Magic_string) anti-pattern.

Reply:
This should probably be addressed by another issue and therefore another branch, but perhaps we should consider modifying the Container component to render a Panel object and encapsulate the logic specific to the panel being created. this could be done by passing a function in the Container constructor which would contain this logic. This would avoid a lot of code duplication. The componentName passed to Container in the constructor is the same as the title used in the panel container and I think the same variable (component_name) is also used elsewhere. the logic could be passed through the props function and defined in the Container component itself.

Reply:
Since we're working in React, I recommend we operate through that rather than passing values to constructors. Something like `this.component.name` will break in production (we should move eventually to separate dev and production code) once it is minified in something like webpack. React has HOCs for this use case: https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging

Reply:
```javascript
class Container extends React.Component {
  constructor(props, componentName, componentLogic) {
    super(props);
    this.componentName = componentName;
    this.refreshData = this.refreshData.bind(this);
    this.state = {
      data: null,
    }
  }

 render () {
return (
<div className= {componentName}>
    <Panel 
        title={componentName}
        refreshData={this.refreshData} >
             /* inserLogic here */
    </Panel>
  componentDidMount() {
    /*does stuff*/
  }

  fetchData() {
       /*does stuff*/
    });
  }

  refreshData() {
    /*does stuff*/
  }
}
```

something along the lines of this

Reply:
this way we could just create a github container by writing

```javascript

const githubLogic = () => {
    /* does some cool stuff bro */
};

const github = new Container(props, 'github', githubLogic);
```

Reply:
fixed